### PR TITLE
Improve Scala 3 evaluation: static objects and private fields

### PR DESCRIPTION
**Pull request by adpi2**
- Add more Scala 3 evaluation tests for `@main` annotations and brace-less syntax
- Fix cast in Scala 3 evaluation of idents
- Refactor Scala 2 evaluator (move phases to different files)
- Add failing tests on accessing static objects and their fields
- Fix evaluation of static objects and private fields

### CODE REVIEW

File: core/src/test/scala/ch/epfl/scala/debugadapter/MoreScala3EvaluationTests.scala [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      assertInMainClass(source, "EvaluateTest")(
+        Breakpoint(12)(
+          ExpressionEvaluation.success("msg(\"Alice\")", "Hello, Alice!"),
+          ExpressionEvaluation.success("msg1(1)", "Hello, 1!"),
+          ExpressionEvaluation.success("msg2(new Foo)", "Hello, foo!")
+        )
+      )
+    }
+
+    "brace-less syntax: evaluate tail-rec function" - {
```
Comment (tgodzik) [inline core/src/test/scala/ch/epfl/scala/debugadapter/MoreScala3EvaluationTests.scala:220]: Btw. could you add a test that has some generic type in it? It seems that evaluator breaks when we get something like `: List[String]`. IF it's not fixed by this PR I can tak a stab at it.

File: expression-compiler/src/main/scala-2/scala/tools/nsc/EvaluationBridge.scala [hunk 2]
```diff
@@ -23,18 +21,17 @@ final class EvaluationBridge {
       timeoutMillis: Long
   ): Boolean = {
     val settings = new Settings
+    // settings.Xprint.value = List("insert-expression", "typer", "generate-expression")
```
Comment (tgodzik) [inline expression-compiler/src/main/scala-2/scala/tools/nsc/EvaluationBridge.scala:24]: That might be useful for debugging, right? Maybe add something like `// this code can be used to print the tree after each phase of the debugger`

File: core/src/test/scala/ch/epfl/scala/debugadapter/MoreScala3EvaluationTests.scala (hunk 1 above)
Comment (adpi2) [reply core/src/test/scala/ch/epfl/scala/debugadapter/MoreScala3EvaluationTests.scala:220]: I fixed that and added a test in `ScalaEvaluationTests`

File: core/src/test/scala/ch/epfl/scala/debugadapter/ScalaEvaluationTests.scala [hunk 3]
```diff
@@ -0,0 +1,876 @@
...
+          |""".stripMargin
+      val evaluation =
+        ExpressionEvaluation.success("List(1, 2, 3).map(_ * a * b * c).sum", 36)
+      assertInMainClass(source, "EvaluateTest")(
+        Breakpoint(6)(evaluation),
+        Breakpoint(15)(evaluation)
+      )
+    }
+
+    "evaluate call to anonymous function" - {
```
Comment (adpi2) [inline core/src/test/scala/ch/epfl/scala/debugadapter/ScalaEvaluationTests.scala:807]: @tgodzik this one tests the evaluation of the generic local value `f`

File: expression-compiler/src/main/scala-2/scala/tools/nsc/EvaluationBridge.scala (hunk 2 above)
Comment (adpi2) [reply expression-compiler/src/main/scala-2/scala/tools/nsc/EvaluationBridge.scala:24]: Ok I'll add this comment in a future PR

**Pull request merged**